libs/sdk-core/src/models.rs

```diff
@@ -132,63 +142,276 @@ pub struct ReverseSwapPairInfo {
 }
 
 /// Details of past or ongoing reverse swaps, as stored in the Breez local DB
-#[derive(Clone, Debug, Serialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct ReverseSwapInfo {
-    // static immutable data
-    pub claim_address: String,
-    pub lockup_address: String,
+    /// The reverse swap ID, as reported by the Boltz API in case of a successful creation
+    pub id: String,
 
-    // dynamic data
-    pub status: ReverseSwapStatus,
+    /// Local timestamp when the reverse swap is created
+    pub created_at: i64,
+
+    /// Locally generated preimage, revealed in the last step of the reverse swap
+    pub local_preimage: Vec<u8>,
+
+    /// Locally generated private key, used to sign the claim tx
+    pub local_private_key: Vec<u8>,
+
+    /// On-chain destination address, to which the reverse swap will finally send funds to
+    pub destination_address: String,
+
+    pub timeout_block_height: u32,
+
+    // Only available in the first response from Boltz API:
+    pub hodl_bolt11: String,
+    pub redeem_script: String,
+
+    /// Amount of sats that will be locked.
+    ///
+    /// The final amount sent will be this value minus the claim tx fees.
+    pub onchain_amount_sat: u64,
+
+    pub cache: ReverseSwapInfoCached,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ReverseSwapInfoCached {
+    /// Reverse swap status, as reported by the Boltz API. Updated on every [crate::BreezServices::sync] for monitored states.
+    pub boltz_api_status: BoltzApiReverseSwapStatus,
+    pub breez_status: ReverseSwapStatus,
+}
+
+impl ReverseSwapInfo {
+    /// Dynamically determine the status, based on the cached Boltz API status.
+    ///
+    /// In case of [BoltzApiReverseSwapStatus::SwapCreated], the payment status is looked up on the LN node,
+    /// to distinguish between a reverse swap that is still ongoing and one where the HODL payment failed.
+    pub(crate) async fn get_dynamic_breez_status(
+        &self,
+        persister: Arc<SqliteStorage>,
+    ) -> Result<ReverseSwapStatus> {
+        let status = match &self.cache.boltz_api_status {
+            BoltzApiReverseSwapStatus::SwapCreated => {
+                let invoice = parse_invoice(&self.hodl_bolt11)?;
+                match persister.get_payment_by_hash(&invoice.payment_hash)? {
```

Comment:
Does this behave right on restart as well? if we paid a hodl invoice, synced and the payment is pending in the db, then restarted. In this case after the initial sync the payment is pending immediately? Just want to make sure that greenlight doesn't have some time to detect this pending payment otherwise we might mark this payment as canceled while there is still ongoing payment.

Reply:
Should be the case, but I only tested and confirmed the failure case, because the initial HODL payment fails with the current mempool fees.

Will try again and post here if this works in the successful case too.